frontend/src/components/MapView/Layers/AnticipatoryActionLayer/index.tsx

```diff
@@ -103,46 +103,62 @@ function AnticipatoryActionLayer({ layer, before }: LayersProps) {
     );
   }, [aaCategories, aaWindow, dispatch, layer.csvWindowKey, selectedDateData]);
 
-  const layers = Object.entries(renderedDistricts[layer.csvWindowKey])
-    .map(([district, { category, phase }]: [string, any]) => {
-      const color = getAAColor(category, phase, true);
-      const icon = getAAIcon(category, phase, true);
-      const features = [
-        data?.features.find(
-          cell =>
-            cell.properties?.[boundaryLayer.adminLevelLocalNames[1]] ===
-            district,
-        ),
-      ];
-      const centroid = districtCentroids[district] || {
-        geometry: { coordinates: [0, 0] },
-      };
-
-      return {
-        id: `anticipatory-action-${district}`,
-        district,
-        data: { ...data, features },
-        color,
-        icon,
-        centroid,
-      };
-    })
-    .filter(x => x.color !== null && x.data.features !== undefined);
+  const coloredDistrictsLayer = React.useMemo(() => {
+    const districtEntries = Object.entries(
+      renderedDistricts[layer.csvWindowKey],
+    );
+    if (!data || !districtEntries.length) {
+      return [];
+    }
+    return {
+      ...data,
+      features: Object.entries(renderedDistricts[layer.csvWindowKey])
+        .map(([districtId, { category, phase }]: [string, any]) => {
+          const feature = data?.features.find(
+            f =>
+              f.properties?.[boundaryLayer.adminLevelLocalNames[1]] ===
+              districtId,
+          );
+
+          if (!feature) {
+            return null;
+          }
+          const color = getAAColor(category, phase, true);
+          return {
+            ...feature,
+            properties: {
+              ...feature.properties,
+              fillColor: color || 'grey',
+            },
+          };
+        })
+        .filter(f => f !== null),
+    };
+  }, [data, renderedDistricts, layer.csvWindowKey]);
```

Comment:
I am not sure I understand why we need to use layer.csvWindowKey in here?
Why are we not simply returning the final prioritized list of district with their max level in `renderedDistricts` ?
It feels like that's what's creating the issue here as `layer.csvWindowKey` seems to be out of sync